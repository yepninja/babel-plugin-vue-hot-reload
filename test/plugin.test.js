const fs = require('fs')
const path = require('path')
const babel = require('@babel/core')

const checkSnap = (file) => {
  const {code} = babel.transformFileSync(path.resolve(__dirname, file))
  expect(code).toMatchSnapshot()
}

test('class component with decorator', () => {
  checkSnap('../mocks/class-with-decorator+jsx+render.tsx')
});

test('object component', () => {
  checkSnap('../mocks/object-with-jsx+render.tsx')
});

test('object component', () => {
  checkSnap('../mocks/class-tsx-without-render.tsx')
});