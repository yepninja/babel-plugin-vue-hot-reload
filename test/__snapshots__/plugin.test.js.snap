// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class component with decorator 1`] = `
"import Vue from 'vue';
import { Component } from 'vue-property-decorator';

@Component
class Logo extends Vue {
  render() {
    return <div class=\\"VueToNuxtLogo\\">
        <div class=\\"Triangle Triangle--one\\" />
      </div>;
  }

}

var __component__ = Logo;
export default __component__;

if (module.hot) {
  var options = typeof __component__ === 'function' ? __component__.options : __component__;

  var api = require('vue-hot-reload-api');

  api.install(require('vue'));

  if (api.compatible) {
    module.hot.accept();

    if (!module.hot.data) {
      api.createRecord('48ed69eb', options);
    } else {
      api.reload('48ed69eb', options);
    }
  }
}"
`;

exports[`object component 1`] = `
"import Vue from 'vue';
var __component__ = {
  render() {
    return <div class=\\"VueToNuxtLogo\\">
        <div class=\\"Triangle Triangle--one\\" />
      </div>;
  }

};
export default __component__;

if (module.hot) {
  var options = typeof __component__ === 'function' ? __component__.options : __component__;

  var api = require('vue-hot-reload-api');

  api.install(require('vue'));

  if (api.compatible) {
    module.hot.accept();

    if (!module.hot.data) {
      api.createRecord('61ebb64a', options);
    } else {
      api.reload('61ebb64a', options);
    }
  }
}"
`;

exports[`object component 2`] = `
"import Vue from 'vue';
import { Component } from 'vue-property-decorator';
export default @Component
class Logo extends Vue {
  render1() {
    return <div class=\\"VueToNuxtLogo\\">
        <div class=\\"Triangle Triangle--one\\" />
      </div>;
  }

}"
`;
